compiler/rustc_const_eval/src/transform/check_consts/resolver.rs

```diff
@@ -78,6 +83,29 @@ where
             self.assign_qualif_direct(&return_place, qualif);
         }
     }
+
+    fn address_of_allows_mutation(&self, mt: mir::Mutability, place: mir::Place<'tcx>) -> bool {
+        match mt {
+            mir::Mutability::Mut => true,
+            mir::Mutability::Not => self.shared_borrow_allows_mutation(place),
```

Comment:
Do you think we should change this to be more conservative here? The semantics is inherited from previously used `MaybeMutBorrowedLocals`.

Reply:
Yeah I think it might be good not to assume that `addr_of!(x.field)` is read-only, for now. I personally think it should be read-only but I am not sure if there is consensus for that.